daemon/graphdriver/overlay2/overlay.go

```diff
@@ -150,11 +161,33 @@ func Init(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (grap
 		ctr:     graphdriver.NewRefCounter(graphdriver.NewFsChecker(graphdriver.FsMagicOverlay)),
 	}
 
-	return d, nil
-}
+	if backingFs == "xfs" {
+		// Try to enable project quota support over xfs.
+		if d.quotaCtl, err = graphdriver.NewQuotaCtl(home); err == nil {
+			//
+			// Test if file system supports project quotas by trying to set a quota
```

Comment:
Ideally this check would be part of `NewQuotaCtl` and done before `getMinProjectID` is called. My worry is that for xfs user who aren't using this feature, they will see a slower start-up linear to the number of layers they have. Do you have any numbers of how long the scan takes?